### PR TITLE
[Deepseek v3.2] Remove extra logics in indexer

### PR DESCRIPTION
**Pull request by IwakuraRein**
## Purpose

Remove extra logics around `index_end_pos` in the indexer.

CC @dcampora 

## Test Plan

`lm-eval --model local-completions --tasks gsm8k --model_args model=DeepSeek-V3.2-Exp,base_url=http://127.0.0.1:8000/v1/completions,num_concurrent=50,max_retries=3,tokenized_requests=False --num_fewshot 20`

## Test Result

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|    20|exact_match|↑  |0.9553|±  |0.0057|
|     |       |strict-match    |    20|exact_match|↑  |0.9560|±  |0.0056|

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: csrc/sampler.cu [hunk 1]
```diff
@@ -113,7 +115,16 @@ static __global__ void topKPerRow(const float* logits, const int* rowStarts,
   // The row computed by this block.
   int rowIdx = blockIdx.x;
   // The range of logits within the row.
-  int rowStart = rowStarts[rowIdx], rowEnd = rowEnds[rowIdx];
+  int rowStart, rowEnd;
+  if (seq_lens == nullptr) {
```
Comment (dcampora) [inline csrc/sampler.cu]: Make all variables in C++ camel case.

File: csrc/sampler.cu [hunk 2]
```diff
@@ -326,6 +337,22 @@ void apply_repetition_penalties_(
...
+                          int64_t next_n, const torch::Tensor& seq_lens,
+                          torch::Tensor& indices, torch::Tensor& values,
+                          int64_t numRows, int64_t stride0, int64_t stride1) {
+  // Compute the results on the device.
+  constexpr int kNumThreadsPerBlock = 512;
+  const cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+
+  vllm::topKPerRow<kNumThreadsPerBlock>
+      <<<numRows, kNumThreadsPerBlock, 0, stream>>>(
+          logits.data_ptr<float>(), nullptr, nullptr, seq_lens.data_ptr<int>(),
```
Comment (dcampora) [inline csrc/sampler.cu]: Please make two global functions instead, with different signatures, and convert the current global into a `device` function that takes the `rowIdx`, `rowStart` and `rowEnd` as variables. Ideally, no `nullptr` or `-1` is needed in any fn invocation.

**Review by dcampora — APPROVED** [review]
LGTM

**Review by heheda12345 — APPROVED** [review]
LGTM! Thank you very much.
